### ISSUE TITLE
TinyMCE conflicts with shadowDOM polyfill

### ISSUE THREAD

**Issue by dgrammatiko** [tinymce/tinymce#4502]
**Do you want to request a *feature* or report a *bug*?**
BUG
**What is the current behavior?**
Editor will not initialise on Firefox
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**
https://codepen.io/dgrammatiko/pen/djNdGz
**What is the expected behavior?**
Editor should initialise normally
**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
TinyMCE 4.0.8, MacOS, Firefox 61

**Comment by fyrkant** [comment]
I assume you mean version 4.8.0 of tinymce? 

Seems like it is the shady-dom polyfill that breaks the editor, hmm I wonder why...

**Comment by dgrammatiko** [comment]
@fyrkant yes, that was a typo the version I'm referring is 4.8.0

**Comment by fyrkant** [comment]
Looking at the shadydom there seems to be other people reporting stuff breaking when using with together with shadydom. Is there any alternative polyfill that you can use instead?

**Comment by dgrammatiko** [comment]
Unfortunately for shadow DOM this is the only polyfill :(

**Comment by fyrkant** [comment]
Ok, I'll dig a little deeper and see if I get any closer to understanding what's breaking.

**Comment by dgrammatiko** [comment]
Thank you!

**Comment by fyrkant** [comment]
I think I found what the problem is and it should be a simple fix that will probably be in the next version of TinyMCE!

**Comment by Magnytu2** [comment]
Hello, I encounter the same problem on Safari.

**Comment by infograf768** [comment]
@fyrkant 
Where do we stand here? Works fine in Safari, not in Firefox (joomla 4.0-dev)

**Comment by fyrkant** [comment]
@infograf768 We have a fix in the pipeline, not sure when it will be released, probably in a few weeks.

**Comment by dgrammatiko** [comment]
@fyrkant I did some debugging and here's what I found:
These are the options we pass to tinyMCE 
```js
{
  "setCustomDir": "",
  "mediaUploadPath": "",
  "uploadUri": "http://joomla.local/administrator/index.php?option=com_media&task=file.upload&tmpl=component&e8ff011987a6c09131f43730c3d47fc6=hampem9vmqthr8ivvnpt69idht&1c74ff13a9b45e1540b4bf4df990a709=1&asset=image&format=json",
  "suffix": ".min",
  "baseURL": "/media/vendor/tinymce",
  "directionality": "ltr",
  "language": "en",
  "autosave_restore_when_empty": false,
  "skin": "lightgray",
  "theme": "modern",
  "schema": "html5",
  "menubar": "edit insert view format table tools",
  "toolbar1": "bold italic underline strikethrough | alignleft aligncenter alignright alignjustify | styleselect | formatselect fontselect fontsizeselect | searchreplace | bullist numlist | outdent indent | undo redo | link unlink anchor | code | forecolor backcolor | fullscreen | table | subscript superscript | charmap emoticons media hr ltr rtl | cut copy paste pastetext | visualchars visualblocks nonbreaking blockquote template | print preview codesample insertdatetime removeformat jxtdbuttons",
  "toolbar2": null,
  "inline_styles": true,
  "gecko_spellcheck": true,
  "entity_encoding": "raw",
  "verify_html": true,
  "valid_elements": "",
  "extended_valid_elements": "hr[id|title|alt|class|width|size|noshade]",
  "invalid_elements": "script,applet,iframe",
  "relative_urls": true,
  "remove_script_host": false,
  "content_css": "/templates/system/css/editor.css",
  "document_base_url": "/",
  "paste_data_images": true,
  "importcss_append": true,
  "image_title": true,
  "height": "550",
  "width": "",
  "resize": "both",
  "templates": [
    {
      "title": "layout1",
      "description": " ",
      "url": "/media/vendor/tinymce/templates/layout1.html"
    },
    {
      "title": "snippet1",
      "description": " ",
      "url": "/media/vendor/tinymce/templates/snippet1.html"
    }
  ],
  "image_advtab": true,
  "dndEnabled": "1",
  "branding": false,
  "force_br_newlines": false,
  "force_p_newlines": true,
  "forced_root_block": "p",
  "rel_list": [
    {
      "title": "None",
      "value": ""
    },
    {
      "title": "Alternate",
      "value": "alternate"
    },
    {
      "title": "Author",
      "value": "author"
    },
    {
      "title": "Bookmark",
      "value": "bookmark"
    },
    {
      "title": "Help",
      "value": "help"
    },
    {
      "title": "License",
      "value": "license"
    },
    {
      "title": "Lightbox",
      "value": "lightbox"
    },
    {
      "title": "Next",
      "value": "next"
    },
    {
      "title": "No Follow",
      "value": "nofollow"
    },
    {
      "title": "No Referrer",
      "value": "noreferrer"
    },
    {
      "title": "Prefetch",
      "value": "prefetch"
    },
    {
      "title": "Prev",
      "value": "prev"
    },
    {
      "title": "Search",
      "value": "search"
    },
    {
      "title": "Tag",
      "value": "tag"
    }
  ],
  "toolbar_items_size": "small",
  "joomlaMergeDefaults": true,
  "joomlaExtButtons": {
    "names": [
      {
        "name": "Article",
        "href": "http://joomla.local/administrator/index.php?option=com_content&amp;view=articles&amp;layout=modal&amp;tmpl=component&amp;1c74ff13a9b45e1540b4bf4df990a709=1&amp;editor=jform_articletext",
        "id": "jform_articletext_Article",
        "icon": "none icon-file-add",
        "click": null
      },
      {
        "name": "Contact",
        "href": "http://joomla.local/administrator/index.php?option=com_contact&amp;view=contacts&amp;layout=modal&amp;tmpl=component&amp;1c74ff13a9b45e1540b4bf4df990a709=1&amp;editor=jform_articletext",
        "id": "jform_articletext_Contact",
        "icon": "none icon-address",
        "click": null
      },
      {
        "name": "Image",
        "href": "http://joomla.local/administrator/index.php?option=com_media&amp;tmpl=component&amp;e_name=jform_articletext&amp;asset=94&amp;author=626",
        "id": "jform_articletext_Image",
        "icon": "none icon-pictures",
        "click": null
      },
      {
        "name": "Menu",
        "href": "http://joomla.local/administrator/index.php?option=com_menus&amp;view=items&amp;layout=modal&amp;tmpl=component&amp;1c74ff13a9b45e1540b4bf4df990a709=1&amp;editor=jform_articletext",
        "id": "jform_articletext_Menu",
        "icon": "none icon-share-alt",
        "click": null
      },
      {
        "name": "Module",
        "href": "http://joomla.local/administrator/index.php?option=com_modules&amp;view=modules&amp;layout=modal&amp;tmpl=component&amp;editor=jform_articletext&amp;1c74ff13a9b45e1540b4bf4df990a709=1",
        "id": "jform_articletext_Module",
        "icon": "none icon-file-add",
        "click": null
      },
      {
        "name": "Page Break",
        "href": "http://joomla.local/administrator/index.php?option=com_content&amp;view=article&amp;layout=pagebreak&amp;tmpl=component&amp;e_name=jform_articletext",
        "id": "jform_articletext_PageBreak",
        "icon": "none icon-copy",
        "click": null
      },
      {
        "name": "Read More",
        "href": null,
        "id": "jform_articletext_ReadMore",
        "icon": "none icon-arrow-down",
        "click": "insertReadmore('jform_articletext');return false;"
      }
    ]
  },
  "selector": null,
  "target": {}
}
```

and this is the way we render:
```js
 // Create a new instance
var ed = new tinyMCE.Editor(element.id, options, tinymce.EditorManager);
ed.render();
```

If I disble the plugins everything works: 
```js
// @todo re enable the plugins
delete options.external_plugins;
delete options.plugins;
```

**Comment by fyrkant** [comment]
The fix for this has now been released, can you all try out TinyMCE 4.8.2 and see if it's working?

**Comment by dgrammatiko** [comment]
Works here, thank you!
